### PR TITLE
Add composite rule

### PR DESCRIPTION
**Pull request by dehbka**
| Q             | A
| ------------- | ---
| Is bugfix?    | ❌
| New feature?  | ✔️
| Breaks BC?    | ❌
| Fixed issues  | [234](https://github.com/yiisoft/validator/issues/234)

### CODE REVIEW

**Review by samdark — APPROVED** [review]

**Comment by xepozz** [comment]
Have you checked rules that already exist? Here we have GroupRule that can solve your possible case.

**Review by vjik — COMMENTED** [review]
Seems, we can not move prevalidate logic from `Validator`, and inherit `Composite` from `GroupRule`.

**Comment by vjik** [comment]
> Have you checked rules that already exist? Here we have GroupRule that can solve your possible case.

Abstract `GroupRule` need for create class with nested rules in userland and then using him.

`Composite` need for direct creation with rules in constructor.

File: src/Rule/CompositeHandler.php [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    public function validate(mixed $value, object $rule, ?ValidationContext $context = null): Result
+    {
+        if (!$rule instanceof Composite) {
+            throw new UnexpectedRuleException(Composite::class, $rule);
+        }
+
+        $context->setParameter($this->parameterPreviousRulesErrored, true);
+
+        if ($this->preValidate($value, $context, $rule)) {
+            return  new Result();
```
Comment (rustamwin) [inline src/Rule/CompositeHandler.php:48]:
```suggestion
            return new Result();
```

**Review by rustamwin — APPROVED** [review]

**Comment by samdark** [comment]
👍

**Pull request merged**